src/scenic/simulators/metadrive/simulator.py

```diff
@@ -0,0 +1,289 @@
+            )
+            obj.metaDriveActor = metaDriveActor
+            return
+
+        # If the object type is unsupported, raise an error
+        raise SimulationCreationError(
+            f"Unsupported object type: {type(obj)} for object {obj}."
+        )
+
+    def executeActions(self, allActions):
+        """Execute actions for all vehicles in the simulation."""
+        super().executeActions(allActions)
+
+        # Apply control updates to vehicles and pedestrians
+        for obj in self.scene.objects[1:]:  # Skip ego vehicle (it is handled separately)
+            if obj.isVehicle:
+                action = obj._collect_action()
+                obj.metaDriveActor.before_step(action)
+                obj._reset_control()
+            else:
+                # For Pedestrians
+                if obj._walking_direction is None:
+                    obj._walking_direction = utils.scenicToMetaDriveHeading(obj.heading)
+                if obj._walking_speed is None:
+                    obj._walking_speed = obj.speed
+                direction = Vector(
+                    math.cos(obj._walking_direction), math.sin(obj._walking_direction)
+                )
+                obj.metaDriveActor.set_velocity(
+                    [direction.x, direction.y], obj._walking_speed
+                )
+
+    def step(self):
+        start_time = time.monotonic()
+
+        # Special handling for the ego vehicle
+        ego_obj = self.scene.objects[0]
+        action = ego_obj._collect_action()
+        self.client.step(action)  # Apply action in the simulator
+        ego_obj._reset_control()
+
+        # Render the scene in 2D if needed
+        if self.render and not self.render3D:
+            self.client.render(
+                mode="topdown", semantic_map=True, film_size=self.film_size, scaling=5
+            )
+
+        # If real-time synchronization is enabled, sleep to maintain real-time pace
+        if self.real_time:
+            end_time = time.monotonic()
+            elapsed_time = end_time - start_time
+            if elapsed_time < self.timestep:
+                time.sleep(self.timestep - elapsed_time)
+
+    def destroy(self):
+        if self.client and self.client.engine:
+            object_ids = list(self.client.engine._spawned_objects.keys())
+            if object_ids:
+                self.client.engine.agent_manager.clear_objects(object_ids)
+            self.client.close()
+
+        super().destroy()
+
+    def getProperties(self, obj, properties):
+        metaDriveActor = obj.metaDriveActor
+        position = utils.metadriveToScenicPosition(
+            metaDriveActor.position, self.scenic_offset
+        )
+        md_ang_vel = metaDriveActor.body.getAngularVelocity()
+        velocity = Vector(*metaDriveActor.velocity, 0)
+        speed = metaDriveActor.speed
+        angularSpeed = 0
+        angularVelocity = Vector(md_ang_vel[0], md_ang_vel[1], md_ang_vel[2])
```

Comment:
```suggestion
        angularVelocity = Vector(*metaDriveActor.body.getAngularVelocity())
```
For consistency with the velocity (also getting rid of `md_ang_vel` above). If you prefer to keep `md_ang_vel` that's fine, but then I'd write `*md_ang_vel` here rather than pulling out all 3 coordinates explicitly, as the intent will be clearer.

Reply:
Updated in commit a9d0d007